test/common/performance/image_helpers.go

```diff
@@ -128,14 +132,46 @@ func StartPerformanceImage(factory sender.LoadGeneratorFactory, typeExtractor re
 			panic(err)
 		}
 
-		waitingExecutors.Add(1)
-		go func() {
-			aggr.Run(ctx)
-			waitingExecutors.Done()
-		}()
+		execs = append(execs, aggr)
 	}
 
-	waitingExecutors.Wait()
+	// wait until all pods are ready
+	ns := testNamespace()
+	log.Printf("Waiting for all Pods to be ready in namespace %s", ns)
+	if err := waitForPods(ns); err != nil {
+		panic(fmt.Errorf("timeout waiting for Pods readiness in namespace %s: %v", ns, err))
+	}
+
+	log.Printf("Starting %d executors", len(execs))
+
+	common.Executors(execs).Run(ctx)
 
 	log.Println("Performance image completed")
 }
+
+func testNamespace() string {
+	if pn := os.Getenv(podNamespaceEnvVar); pn != "" {
+		return pn
+	}
+	return defaultTestNamespace
+}
+
+func waitForPods(namespace string) error {
+	c, err := pkgtest.NewKubeClient("", "")
+	if err != nil {
+		return err
+	}
+
+	p := c.Kube.CoreV1().Pods(namespace)
```

Comment:
I replaced that method to avoid failure when when a failure happen while dialing the apiserver. I used to have problems with this particular issue, but i think now i can revert it

Reply:
:+1: If we don't have that problem anymore, let's use the shared code.